src/chess/piece/Pawn.java

```diff
@@ -1,24 +1,81 @@
 package chess.piece;
 
-import java.util.HashMap;
-import java.util.Map;
+import chess.Position;
+
+import java.util.*;
 
 public class Pawn extends Piece {
   private static final Map<Color, Character> symbols = new HashMap<Color, Character>(){{
     put(Color.WHITE, '♙');
     put(Color.BLACK, '♟');
   }};
 
+  // if this pawn is moved, isFirstMoved turn out true but at next your own turn, it turn out false
+  // this variable is related to "en passant"
+  // ex. n:false -> n + 1:true -> n + 2:false
+  private boolean isFirstMoved;
+
+  // if this pawn is moved, isMoved turn out true and it never change again
+  // when the value is true, this Pawn can move forward two squares.
+  // ex. n:false -> n + 1:true
+  private boolean isMoved;
+
   public Pawn(Color color) {
     super(color, symbols);
+    this.isFirstMoved = false;
+    this.isMoved = false;
   }
 
   @Override
-  public boolean isValidMove(Position newPosition) {
-    if (!super.isValidMove(newPosition)) {
+  public boolean isValidMove(int row, int col, boolean isEnemyPiece) {
+    if (!super.isValidMove(row, col, isEnemyPiece)) {
       return false;
     }
-    // ToDo: implement
-    return true;
+
+    // if arguments are impossible int
+    // row = 1 or 2, col = 0 or 1
+    if (row == 0 || 3 <= row || 2 <= col) return false;
```

Comment:
The order of values on a comparison is depending on people, but I feel majority put "comparison base (most of the cases, it's variable)" on left, so this line might be better to be `if (row == 0 || row >= 3 || col >= 2) return false;`
When we explain this part in English it'll be like "if `row` is equal to 0 or `row` is greater or equal to 3 or `col` greater or equal to 2", so the code I mentioned before is more direct

I actually couldn't find any part mentioning about this rule in Google Java Style Guide
instead, I found the article(written in Japanese) which is the summary of "The Art of Readable Code" and it suggests the same way
https://qiita.com/AKB428/items/20e81ccc8d9998b5535d#if%E3%81%AE%E4%B8%A6%E3%81%B3%E9%A0%86%E3%81%AE%E6%8C%87%E9%87%9D

When you want to mean for example "X is between 1 and 10", you might write `1 < X && x < 10`
I personally can accept this and this is only time I change this rule

but again, it depends on developers

Reply:
I felt this code is a bit verbose(in other words, I think this is unnecessary)